code/game/objects/effects/decals/cleanable/humans.dm

```diff
@@ -68,7 +71,7 @@
 	if(rename)
 		name = "rotting [initial(name)]"
 		desc += " They smell terrible."
-	AddComponent(/datum/component/rot/gibs)
+	SEND_SIGNAL(src, COMSIG_START_MIASMA_GENERATION)
```

Comment:
```suggestion
	SEND_SIGNAL(src, COMSIG_START_MIASMA_GENERATION, rename)
```
Add signals to the start of the proc whenever possible, past the normal return conditions, and make them pass the arguments of the proc even if it's currently not useful.

Reply:
I'm essentially using the signal as a proc call to the component here, which is kinda ick, and why most of this code is catching your eye. I'm not too experienced with component development, could I get some advice on this?

Reply:
Signals are very much not meant to replace proc calls. They are event reporters, and should be unique.

Reply:
How do you recommend implementing this properly?

Reply:
Consider this: starting to rot is an event.
Is the gibs rotting essentially different from the corpse's rotting?
If so, make two different unique signals for each, and register to both on the component.
If not, make a common `start_rotting` proc for both that sends the signal and handles both cases generically.

Reply:
Is it really that simple?